src/plugins/saved_objects/README.md

```diff
@@ -0,0 +1,119 @@
+# Saved object
+
+The saved object plugin provides all the core services and functionalities of saved objects. It is utilized by many core plugins such as [`visualizaton`](../visualizations/), [`dashboard`](../dashboard/) and [`wizard`](../wizard/), as well as external plugins. Saved object is the primary way to store app and plugin data in a standardized form in OpenSearch Dashboards. They allow plugin developers to manage creating, saving, editing and retrieving data for the application. They can also make reference to other saved objects and have useful features out of the box, such as migrations and strict typings. The saved objects are managed by the Saved Object Management UI.
+
+## Save relationships to index pattern
+
+Saved objects that have relationships to index patterns are saved using the [`kibanaSavedObjectMeta`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/4a06f5a6fe404a65b11775d292afaff4b8677c33/src/plugins/saved_objects/public/saved_object/helpers/serialize_saved_object.ts#L59) attribute and the [`references`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/4a06f5a6fe404a65b11775d292afaff4b8677c33/src/plugins/saved_objects/public/saved_object/helpers/serialize_saved_object.ts#L60) array structure. Functions from the data plugin are used by the saved object plugin to manage this index pattern relationship. 
+
+A standard saved object and its index pattern relationship:
+
+```ts
+
+"kibanaSavedObjectMeta" : {
+    "searchSourceJSON" : """{"filter":[],"query":{"query":"","language":"kuery"},"indexRefName":"kibanaSavedObjectMeta.searchSourceJSON.index"}"""
+      }
+    },
+    "type" : "visualization",
+    "references" : [
+      {
+          "name" : "kibanaSavedObjectMeta.searchSourceJSON.index",
+          "type" : "index-pattern",
+          "id" : "90943e30-9a47-11e8-b64d-95841ca0b247"
+      }
+    ],
+
+```
+
+### Saving a saved object
+
+When saving a saved object and its relationship to the index pattern: 
+
+1. A saved object will be built using [`buildSavedObject`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/4a06f5a6fe404a65b11775d292afaff4b8677c33/src/plugins/saved_objects/public/saved_object/helpers/build_saved_object.ts#L46) function. Services such as hydrating index pattern, initializing and serializing the saved object are set, and configs such as saved object id, migration version are defined.
```

Comment:
What does hydrating index pattern mean

Reply:
The function ensures that the searchSource.index is a IndexPattern object and it will return that IndexPattern object; otherwise, it will return null.